src/useFormState.ts

```diff
@@ -128,6 +139,70 @@ export function useFormState<T, I>(opts: UseFormStateOpts<T, I>): ObjectState<T>
   return form;
 }
 
+export type ObjectStateCache<T, I> = Record<string, [ObjectState<T>, I]>;
+type UseFormStatesOpts<T, I> = {
```

Comment:
Maybe move this to a new `useFormStates.ts` file? Just glancing but it looks pretty self-contained, which is great.